src/f32ext/fract.rs

```diff
@@ -0,0 +1,89 @@
+use super::copysign;
+use super::utils;
+use super::utils::FloatComponents;
+/// Floating point whole number for a single-precision float.
+use core::f32;
+use core::u32;
+
+pub(super) fn fract_sign(x: f32) -> f32 {
+    let x_bits: u32 = x.to_bits();
+    let exponent: i32 = x.extract_exponent_value();
+    // we know it is *only* fraction
+    if exponent < 0_i32 {
+        return x;
+    }
+    let exponent_clamped = i32::max(exponent, 0_i32) as u32;
+
+    // find the part of the fraction that would be left over
+    let fractional_part: u32 = x_bits.overflowing_shl(exponent_clamped).0 & utils::MANTISSA_MASK;
+    // if there isn't a fraction we can just return 0
+    if fractional_part == 0_u32 {
+        // most people don't actually care about -0.0, so would it be better to just not copysign?
+        return copysign::copysign(0.0_f32, x);
+    }
+    //alternatively use -1.0? subtraction would surely be more costly though right?
```

Comment:
This is an example of the sort of question I was referring to. Perhaps this could be:

```
// Note: alternatively this could use -1.0, but it's assumed subtraction would be more costly

Reply:
Okay that makes sense! I've never been sure how to phrase those kinds of weird questions not questions